warehouse/packaging/services.py

```diff
@@ -443,32 +448,36 @@ def _hit_ratelimits(self, request, creator):
         self.ratelimiters["project.create.user"].hit(creator.id)
         self.ratelimiters["project.create.ip"].hit(request.remote_addr)
 
-    def check_project_name(self, name: str) -> ProjectNameUnavailableReason | None:
+    def check_project_name(self, name: str) -> ProjectNameUnavailableError | None:
         if not PROJECT_NAME_RE.match(name):
-            return ProjectNameUnavailableReason.Invalid
+            return ProjectNameUnavailableInvalid()
 
         # Also check for collisions with Python Standard Library modules.
         if canonicalize_name(name) in STDLIB_PROHIBITED:
-            return ProjectNameUnavailableReason.Stdlib
+            return ProjectNameUnavailableStdlib()
 
-        if self.db.query(
-            exists().where(Project.normalized_name == func.normalize_pep426_name(name))
-        ).scalar():
-            return ProjectNameUnavailableReason.AlreadyExists
+        if (
+            existing_project := self.db.query(Project)
+            .where(Project.normalized_name == func.normalize_pep426_name(name))
+            .scalar()
```

Comment:
Hm, what would happen in the event that two conflicting names exist? We didn't always enforce this and did not retro-actively "clean up" so it could happen?

I think in the pre-existing implementation papers over that because the "exists" clause will end up a boolean through the "scalar" clause, where as after I think you'll get a list of Project objects.

Reply:
doesn't `scalar` return the first result?

Reply:
I recall it doing so if one and only one exists, otherwise a multiple results exception.

Would you mind adding a test to validate the behavior if two preexisting conflicting names exist?

Reply:
uh yeah, I was looking at the wrong docs. Good catch! Fixed by replacing `scalar()` with `first()`, and added a test for it.